tritium-registry/src/test/java/com/palantir/tritium/metrics/registry/LockFreeExponentiallyDecayingReservoirTest.java

```diff
@@ -0,0 +1,450 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tritium.metrics.registry;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.codahale.metrics.Clock;
+import com.codahale.metrics.Reservoir;
+import com.codahale.metrics.Snapshot;
+import com.codahale.metrics.Timer;
+import com.codahale.metrics.Timer.Context;
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.junit.jupiter.api.Test;
+
+/**
+ * {@link LockFreeExponentiallyDecayingReservoirTest} is based closely on the codahale
+ * <a href="https://github.com/dropwizard/metrics/blob/a314e903d7ba3c5ec47ee7c9103d73ed09a20bd6/metrics-core/src/test/java/com/codahale/metrics/ExponentiallyDecayingReservoirTest.java">
```

Comment:
:+1: for the links - I guess the alternative approach is to use the codahale version as a 'golden master' reference impl, like we did for the dialogue futures https://github.com/palantir/dialogue/blob/develop/dialogue-futures/src/test/java/com/palantir/dialogue/futures/DialogueFuturesTest.java, but I think I'm OK with this :)